vignettes/adae.Rmd

```diff
@@ -0,0 +1,325 @@
+#---------------------------------------------------------------------------------------
+ae <- convert_blanks_to_na(ae)
+adsl <- convert_blanks_to_na(adsl)
+```
+
+## Step 2: ADAE Derivations
+
+```{r}
+# ADAE derivation start
+# Read in specifications from define
+#---------------------------------------------------------------------------------------
+## placeholder for origin=predecessor, use metatool::build_from_derived()
+metacore <- spec_to_metacore("/cloud/project/adam/TDF_ADaM - Pilot 3 Team updated.xlsx",
+  where_sep_sheet = FALSE,
+  quiet = T
+)
+adae_spec <- metacore %>%
+  select_dataset("ADAE") # Get the specifications for the dataset we are currently building
+
+
+# Get list of ADSL vars
+#---------------------------------------------------------------------------------------
+adsl_vars <- vars(
+  TRTSDT,
+  TRTEDT,
+  STUDYID,
+  SITEID,
+  TRT01A,
+  TRT01AN,
+  AGE,
+  AGEGR1,
+  AGEGR1N,
+  RACE,
+  RACEN,
+  SEX,
+  SAFFL,
+  TRTSDT,
+  TRTEDT
+)
+
+# Merge adsl to ae
+#---------------------------------------------------------------------------------------
+adae0 <- ae %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by = vars(STUDYID, USUBJID)
+  ) %>%
+  # Set TRTA and TRTAN from ADSL
+  #-------------------------------------------------------------------------------------
+  rename(
+    TRTA = TRT01A,
+    TRTAN = TRT01AN
+  ) %>%
+  # Derive analysis start time
+  #-------------------------------------------------------------------------------------
+  derive_vars_dtm(
+    dtc = AESTDTC,
+    new_vars_prefix = "AST",
+    highest_imputation = "D"
+  ) %>%
+  # Derive analysis end time
+  #-------------------------------------------------------------------------------------
+  derive_vars_dtm(
+    dtc = AEENDTC,
+    new_vars_prefix = "AEN",
+    highest_imputation = "h",
+    max_dates = NULL
+  ) %>%
+  # Derive analysis start & end dates
+  #-------------------------------------------------------------------------------------
+  derive_vars_dtm_to_dt(vars(ASTDTM, AENDTM)) %>%
+  # Duration of AE
+  #-------------------------------------------------------------------------------------
+  derive_vars_dy(
+    reference_date = TRTSDT,
+    source_vars = vars(TRTSDT, ASTDT, AENDT)
+  ) %>%
+  derive_vars_duration(
+    new_var = ADURN,
+    new_var_unit = ADURU,
+    start_date = ASTDT,
+    end_date = AENDT,
+    out_unit = "days"
+  ) %>%
+  mutate(ADURU = str_replace(ADURU, "DAYS", "DAY")) %>%
+  # Treatment Emergent Analysis flag
+  #-------------------------------------------------------------------------------------
+  restrict_derivation(
+    derivation = derive_var_trtemfl,
+    args = params(
+      start_date = ASTDT,
+      end_date = AENDT,
+      trt_start_date = TRTSDT
+    ),
+    filter = !is.na(ASTDT)
+  ) %>%
+  # AOCCFL - 1st Occurrence of Any AE Flag
+  #-------------------------------------------------------------------------------------
```

Comment:
I guess this might go back to the original code, but can we break this up into intermediate datasets?  To showcase some of the derivations being done here.  I feel like this is where people are going to be very interested.  

I guess we need to make sure these two files stay in sink??

Reply:
> I guess we need to make sure these two files stay in sink??

That is exactly the concern I have with duplicating the code in two places and why I'm not fond of creating these vignettes. The added value compared to just looking at the code on GitHub is little but it comes with quite some downsides.

Reply:
I'm just wondering how we can present our work done in a way that is consumable by others either not familiar with looking at code on GitHub, not familiar with R or even higher-ups looking at adopting R into their business process.  I see this project more about demonstrating the fitness of R to others not using it already, hence my interest in making the pilot more accessible to others.   

Vignettes is one way to do this - We can actually access the `submission/programs` folder and `cat` them to the vignette section so we don't have to maintain two files.  But I am open to other suggestions on how to "sell" the Pilot more

Whatever we decide, we also might want to revisit how the code is laid out on GitHub as each dataset program is slightly different.  The ADAE is one big pipe while ADADAS is broken into modularized pieces.  I prefer ADADAS as I can run the code and see the build up of the pieces - also makes it easier to showcase in a vignette.

Reply:
> Vignettes is one way to do this - We can actually access the submission/programs folder and cat them to the vignette section so we don't have to maintain two files. But I am open to other suggestions on how to "sell" the Pilot more

I had a similar idea with just `cat`ing the scripts in mind. Regardless, I do think properly documenting our work is key to share our experience with various stakeholders. I am, however, not sure whether a pkgdown site is the right way to do so. We're really abusing the package structure in this pilot. We could look at creating a more customized webiste using something like `quarto` or Hugo.

> Whatever we decide, we also might want to revisit how the code is laid out on GitHub as each dataset program is slightly different. The ADAE is one big pipe while ADADAS is broken into modularized pieces. I prefer ADADAS as I can run the code and see the build up of the pieces - also makes it easier to showcase in a vignette.

Making all scripts consistent is something I'm planning to do prior to creating the finl submission package.

Reply:
I agree with @bms63 as well. If we can make the code more modular then we can automate the vignette creation easily.